tests/unit/niceml/data/normalization/test_dataframe.py

```diff
@@ -22,25 +60,184 @@ def sample_dataframe():
     assert norm_info.offset == expected_norm_info.offset
     assert norm_info.divisor == expected_norm_info.divisor
+
+
+@pytest.mark.parametrize(
+    "column_key, expected_output, expected_norm_info",
+    [
+        (
+            "column2",
+            pd.DataFrame({"column1": [1, 2, 3, 4, 5], "column2": [0, 1, 2, 3, 4]}),
+            CategoricalNormalizationInfo(
+                feature_key="column2", values=[10, 20, 30, 40, 50]
+            ),
+        ),
+        (
+            "column1",
+            pd.DataFrame({"column1": [0, 1, 2, 3, 4], "column2": [10, 20, 30, 40, 50]}),
+            CategoricalNormalizationInfo(feature_key="column1", values=[1, 2, 3, 4, 5]),
+        ),
+    ],
+)
+def test_normalize_categorical_col(
+    sample_dataframe, column_key, expected_output, expected_norm_info
+):
+    input_dataframe = sample_dataframe.copy()
+
+    output_dataframe, norm_info = normalize_categorical_column(
+        input_dataframe, column_key
+    )
+
+    assert output_dataframe.equals(expected_output)
+    assert norm_info.feature_key == expected_norm_info.feature_key
+    assert norm_info.values == expected_norm_info.values
+
+
+@pytest.mark.parametrize(
+    "column_key, expected_output, expected_norm_info",
+    [
+        (
+            "column1",
+            pd.DataFrame(
+                {"column1": [1, 0, 1, 1, 0], "column2": ["OK", "OK", "OK", "NOK", "OK"]}
+            ),
+            BinaryNormalizationInfo(feature_key="column1", values=["NOK", "OK"]),
+        ),
+        (
+            "column2",
+            pd.DataFrame(
+                {
+                    "column1": ["OK", "NOK", "OK", "OK", "NOK"],
+                    "column2": [1, 1, 1, 0, 1],
+                }
+            ),
+            BinaryNormalizationInfo(feature_key="column2", values=["NOK", "OK"]),
+        ),
+    ],
+)
+def test_normalize_binary_col(
+    sample_binary_dataframe, column_key, expected_output, expected_norm_info
+):
+    input_dataframe = sample_binary_dataframe.copy()
+
+    output_dataframe, norm_info = normalize_binary_column(input_dataframe, column_key)
+
+    assert output_dataframe.equals(expected_output)
+    assert norm_info.feature_key == expected_norm_info.feature_key
+    assert norm_info.values == expected_norm_info.values
+
+
+@pytest.mark.parametrize(
+    "expected_output,normalization_info",
+    [
+        (
+            pd.DataFrame(
+                {
+                    "column1": ["OK", "NOK", "OK", "OK", "NOK"],
+                    "column2": [1, 1, 1, 0, 1],
+                }
+            ),
+            BinaryNormalizationInfo(feature_key="column1", values=["NOK", "OK"]),
+        ),
+        (
+            pd.DataFrame(
+                {"column1": [1, 0, 1, 1, 0], "column2": ["OK", "OK", "OK", "NOK", "OK"]}
+            ),
+            BinaryNormalizationInfo(feature_key="column2", values=["NOK", "OK"]),
+        ),
+    ],
+)
+def test_denormalize_binary_col(
```

Comment:
Add a test where a dataframe is normalized and then denormalized and should be the same as the input dataframe again. Please add this test for all normalization methods.

Reply:
In my opinion, the tests should be as granular as possible and if possible test only one functionality. There is a test for normalization and denormalization for the existing FeatureType. Or did I miss a functionality/combination?

Reply:
Yes, but in this case the two functions are per definition the opposite of eachother. So I think it would be good to test, if they really are.

Reply:
In my opinion, it is not necessary and bad practice to test two functionalities in a single unit test. Besides, this way there are redundancies in the test cases.
I understand your arguments. The opposite of normalization is denormalization, but it doesn't say that you can only use the denormalization function after calling the normalization function. 
But let's discuss this in person if you want.